### PR TITLE
[fix](serde) fix ipv4/v6 serde functions for arrow, orc, parquet format

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
this PR is from @sjyango work in https://github.com/apache/doris/pull/32326,
wants merge #32326 into master branch, but it's draft  and not maintain long time. so have this new PR.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/data_types/serde/data_type_ipv4_serde.cpp [hunk 1]
```diff
@@ -108,5 +109,29 @@ Status DataTypeIPv4SerDe::read_column_from_pb(IColumn& column, const PValues& ar
     }
     return Status::OK();
 }
+
+void DataTypeIPv4SerDe::write_column_to_arrow(const IColumn& column, const NullMap* null_map,
+                                              arrow::ArrayBuilder* array_builder, int start,
+                                              int end, const cctz::time_zone& ctz) const {
+    const auto& col_data = static_cast<const ColumnIPv4&>(column).get_data();
```
Comment (HappenLee) [inline be/src/vec/data_types/serde/data_type_ipv4_serde.cpp:116]: use assert_cast

File: be/src/vec/data_types/serde/data_type_ipv6_serde.cpp [hunk 2]
```diff
@@ -112,5 +114,88 @@ Status DataTypeIPv6SerDe::read_column_from_pb(IColumn& column, const PValues& ar
     return Status::OK();
 }
 
+void DataTypeIPv6SerDe::write_column_to_arrow(const IColumn& column, const NullMap* null_map,
+                                              arrow::ArrayBuilder* array_builder, int start,
+                                              int end, const cctz::time_zone& ctz) const {
+    const auto& col_data = static_cast<const ColumnIPv6&>(column).get_data();
```
Comment (HappenLee) [inline be/src/vec/data_types/serde/data_type_ipv6_serde.cpp:120]: assert_cast

File: be/src/vec/data_types/serde/data_type_ipv6_serde.cpp [hunk 3]
```diff
@@ -112,5 +114,88 @@ Status DataTypeIPv6SerDe::read_column_from_pb(IColumn& column, const PValues& ar
...
+                             column.get_name(), array_builder->type()->name());
+        }
+    }
+}
+
+void DataTypeIPv6SerDe::read_column_from_arrow(IColumn& column, const arrow::Array* arrow_array,
+                                               int start, int end,
+                                               const cctz::time_zone& ctz) const {
+    auto& col_data = static_cast<ColumnIPv6&>(column).get_data();
+    const auto* concrete_array = dynamic_cast<const arrow::StringArray*>(arrow_array);
```
Comment (HappenLee) [inline be/src/vec/data_types/serde/data_type_ipv6_serde.cpp:138]: assert_cast?

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**